the-collab-lab/tcl-24-smart-shopping-list
# 3 cv am local storage

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



Created Home and Protected Route components.

In Home, a button to create a new list is rendered if there is no token in local storage, if there is, it redirects it to '/list'. When the button is clicked, a new token is issued and saved in local storage.

In App, the list components is wrapped around the protected route component to make sure there is a token in local storage, if not, the user is redirected to '/' home to create a new list (and token).

## Related Issue



Closes #4 

## Acceptance Criteria



- [x] A button should be displayed in home, for users that don't have a token nor list,  that allows them to create a new list.
- [x] Clicking the button generates a new token and saves it in local storage.
- [x]  The user is redirected to the list view once the token has been created and saved. 
- [x] Users that already have a saved token in local storage must be redirected to the list view. 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After



<img width="1440" alt="Screen Shot 2021-04-14 at 5 03 54 PM" src="https://user-images.githubusercontent.com/66270746/114786266-71a5e800-9d43-11eb-82d3-70cc7f868c2b.png">

## Testing Steps / QA Criteria



From your terminal, pull down the branch
Run npm install to install the dependencies
Run npm start to view the features in the browser
Click new list button
Check for redirection to the list route
Check local storage for token
Add a new item to your list 
Check that all the items from your list are being displayed
